File: Advanced_SQL_50/1755-defuse-the-bomb/defuse-the-bomb.go
func decrypt(circ []int, k int) []int {
    n := len(circ)
    result := make([]int, n)
    
    if k == 0 {
        return result
    }
    
    wSum := 0
    var start, end int
    
    if k > 0 {
        start = 1
        end = k
    } else {
        start = n + k
        end = n - 1
    }
    
    for i := start; i <= end; i++ {
        wSum += circ[i%n]
    }
    
    for i := 0; i < n; i++ {
        result[i] = wSum
        wSum -= circ[(start+i)%n]
        wSum += circ[(end+i+1)%n]
    }
    
    return result
}

File: solutions/1755-defuse-the-bomb/defuse-the-bomb.swift
class Solution {
    func decrypt(_ circ: [Int], _ k: Int) -> [Int] {
        let n = circ.count
        var result = Array(repeating: 0, count: n)
        
        if k == 0 { return result }
        
        var wSum = 0
        let start = k > 0 ? 1 : n + k
        let end = k > 0 ? k : n - 1
        
        for i in start...end {
            wSum += circ[i % n]
        }
        
        for i in 0..<n {
            result[i] = wSum
            wSum -= circ[(start + i) % n]
            wSum += circ[(end + i + 1) % n]
        }
        
        return result
    }
}